src/vs/vscode.d.ts

```diff
@@ -1805,6 +1806,12 @@ declare module 'vscode' {
 		 */
 		ignoreFocusOut?: boolean;
 
+		/**
+		 * Set to `true` to show a multiline prompt that inserts a new line when user presses 'Shift + Enter'
+		 * instead of submitting
+		 */
+		multiline?: boolean;
+
```

Comment:
I mean we could have the type as -
```typescript
| { multiline?: false, password?: boolean }
| { multiline?: true, password?: false }
```

Pros - 
1. Impossible to write `showInputBox({ multiline: true, password: true })` for typescript users

Cons -
1. Terrible typescript inference and error message ergonomics if the options are not literal
2. We'll need to throw in case of `{ multiline: true, password: true }` because hey there are JS folks too